FreeRTOS/iot-reference-arm-corstone3xx
# Add tests for and debug mqtt_agent_task.c

Description
-----------
This MR also adds a substantial number (>40) of FFF fakes to be used in future unit testing.

Test Steps
-----------
Run the following from the project's root directory:
`rm -rf build_unit_test && cmake -S . -B build_unit_test -GNinja && cmake --build build_unit_test && ctest --test-dir build_unit_test --output-on-failure`

Checklist:
----------



Related Issue
-----------


By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
